instrumentation/opentelemetry-instrumentation-aiobotocore/tests/test_aiobotocore_dynamodb.py

```diff
@@ -0,0 +1,512 @@
+# Copyright The OpenTelemetry Authors
```

Comment:
as far as i can see the tests are mostly duplicates of the ones in botocore. i'd imagine these tests and the ones in botocore will get out of sync pretty soon once some new functionality is added. 

is there maybe some way to unify them?

Reply:
I was thinking of using inheritance but it probably wouldn't be so elegant.

Do you know of any similar cases that could serve as inspiration?

It seems like ddtrace accepted that level of duplication in tests
- https://github.com/DataDog/dd-trace-py/blob/1.x/tests/contrib/aiobotocore/test.py
- https://github.com/DataDog/dd-trace-py/blob/1.x/tests/contrib/botocore/test.py

Reply:
tbh, i haven't thought about it that deeply yet, but i was thinking in the direction of dynamically generating the test classes from the botocore test classes with some meta class.
something like starting from a common (maybe abstract) base class for the botocore tests, finding the actual botocore test classes from that, and use the meta class to generate an adapted test class for aiobotocore.
the base class would probably need some factory method for creating the botocore client/resource instances and for the aiobotocore instances some wrapping to make the async functions callable like sync functions.

This is just some wild idea and i'm not sure if it will even work, like:
* will the dynamically created tests be detected by the test framework
* are the botocore tests even installed and accessible when installing the aiobotocore instrumentation for testing
* possible nuances in the handling between botocore and aiobotocore which would require different handling in certain test cases
* ...

so i'm not sure if it would be really worth to try out this approach, also since it most likely will make the test code hard to follow and less readable.

duplicating the tests is also fine i gues and if they really get out of sync it shouldn't be a big problem since the AioBotocoreInstrumentor more or less just calls into and reuses the botocore instrumentation.

Reply:
I tried, but couldn't import them as the tests are not included in the library... doing some sort of client proxy that async-ifies `client.something` calls would be possible, but I share the concern that it would affect readability.

Something else to notice though is that aiobotocore will almost always use an older version of botocore underneath, as the patching efforts are a manual process, and often [not trivial](https://github.com/aio-libs/aiobotocore/pull/999#issuecomment-1457485410).